src/main/scala/com/microsoft/hyperspace/actions/VacuumAction.scala

```diff
@@ -49,4 +52,17 @@ class VacuumAction(
       }
     }
   }
+
+  final override protected def event(message: String): HyperspaceEvent = {
+    val sc = SparkSession.getActiveSession.getOrElse {
+      throw HyperspaceException("No spark session found")
+    }.sparkContext
+
+    VacuumActionEvent(
+      sc.sparkUser,
+      sc.applicationId,
+      sc.appName,
```

Comment:
looks like these are repeating. can you try to refactor?

Reply:
changed event() method to take in spark user, appid, app name. Could you please have a look ?